core/node/eth_sender/src/aggregator.rs

```diff
@@ -40,20 +41,33 @@ pub struct Aggregator {
     operate_4844_mode: bool,
     pubdata_da: PubdataDA,
     commitment_mode: L1BatchCommitmentMode,
-    priority_merkle_tree: SyncMerkleTree<L1Tx>,
+    priority_merkle_tree: MiniMerkleTree<L1Tx>,
 }
 
 impl Aggregator {
-    pub fn new(
+    pub async fn new(
         config: SenderConfig,
         blob_store: Arc<dyn ObjectStore>,
         operate_4844_mode: bool,
         commitment_mode: L1BatchCommitmentMode,
-        priority_merkle_tree: SyncMerkleTree<L1Tx>,
-    ) -> Self {
+        connection: &mut Connection<'_, Core>,
+    ) -> anyhow::Result<Self> {
         let pubdata_da = config.pubdata_sending_mode.into();
 
-        Self {
+        let priority_op_hashes =
+            if let Some(priority_tree_start_index) = config.priority_tree_start_index {
```

Comment:
If `priority_tree_start_index` is `None`, the behavior is identical to assuming it is equal to the last `priority_op_id`. A bit counter-intuitive, I would expect it to behave as `.unwrap_or(0)` since for new chains that won't undergo the migration (and hence don't need this config var) that would be the case

Reply:
actually it should work as None=Some(0) in the end because ops starting from id = 0 will be inserted at the first iteration here
```
let new_l1_tx_hashes = storage
                    .transactions_dal()
                    .get_l1_transactions_hashes(self.priority_merkle_tree.length())
                    .await
                    .unwrap();
for hash in new_l1_tx_hashes {
    self.priority_merkle_tree.push_hash(hash);
}
```
And this doesn't work as intended when priority_tree_start_index is Some(x) if x > 0. I will push fixes